docs/en/ingest-management/fleet-agent-proxy-support.asciidoc

```diff
@@ -0,0 +1,310 @@
+
+|===
+
+Proxy settings are valid in the {agent} policy under the following sections:
+
+* `outputs` - set the proxy server for communicating with {es}.
+
+* `fleet` - set the proxy server for retrieving agent policies from
+{fleet}.
+
+* `agent.download` - set the proxy server for retrieving artifacts or binary
+upgrades from `artifacts.elastic.co`.
+
+//TODO: Clarify where to add these settings ^^. For Fleet-managed agents, it
+//sounds like you need to set these in both the UI and fleet.yml, but it seems
+//odd to ask users to configure a file on the host when the policy is managed
+//by Fleet. 
+
+For {fleet}-managed agents, specify proxy settings in the {kib} UI:
+
+. Log in to {kib} and go to *Management > {fleet}*.
+
+. Click *{fleet} settings*.
+
+. Under *Elasticsearch output configuration (YAML)*, specify proxy settings for
+connecting to {es}. The proxy settings you specify here are applied to all
+{agent}s enrolled in {fleet}.
++
+//TODO: Add a realistic example here.
+
+For standalone agents, specify proxy settings the `elastic-agent.yml` file. For
+example:
+
+[source,yaml]
+----
+outputs:
+  default:
+    type: elasticsearch
+    hosts:
+      - 'http://localhost:9200'
+    proxy_url: 'https://192.0.2.0:8080'
+----
+
+For more information, refer to <<elastic-agent-configuration>>.
+
+[discrete]
+[[cli-proxy-settings]]
+=== Command-line flags when enrolling in {fleet}
+
+Specify proxy settings on the command line if you are enrolling in {fleet} and
+the proxy configuration is not yet available. These settings are added to the
+`fleet.yml` file; you'll still need to
+<<fleet-proxy-configuration,configure {fleet} proxy settings>> for {fleet} to
+  access the package registry and to communicate with {es}.
+
+The `enroll` and `install` commands accept the following flags:
+
+|===
+| CLI flag | Description
+
+|`--proxy-url <url>`
+|URL of the proxy server. The value may be either a complete URL or a
+`host[:port]`, in which case the http scheme is assumed.  The URL accepts optional
+username and password settings for authenticating with the proxy. For example:
+`http://<username>:<password>@<proxy host>/`.
+
+|`--proxy-disabled`
+|If specified, all proxy settings, including the `HTTP_PROXY` and `HTTPS_PROXY`
+environment variables, are ignored.
+
+|`--proxy-header <header name>=<value>`
+|Additional header to send to the proxy during CONNECT requests. Use the
+`--proxy-header` flag multiple times to add additional headers. You can use
+this setting to pass keys/tokens required for authenticating with the proxy.
+
+|===
+
+For example:
+
+[source,sh]
+----
+//TODO: Add an example here
+----
+
+NOTE: These commands require default policies to be loaded in {fleet}. Default
+policies are loaded automatically when you visit {fleet} for the first time. If
+you're not sure whether default policies are loaded, log in to {kib} and go to
+*Management > {fleet}*.
+
+[discrete]
+[[fleet-proxy-configuration]]
+== Configure {fleet} proxy settings
```

Comment:
This is about the Fleet app in Kibana? Might help to be more explicit in the title

Reply:
the fleet-app in kibana can be used to set the output proxy (i.e., `agent->proxy->es`)

Reply:
Marking this as "resolved" because my next commit will include a lot of changes to this section in response to your comments.